apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -21,20 +21,31 @@ export const getFolders = () =>
     .execute()
 
 // NOTE: Base method for retrieving a resource - no distinction made on whether `blobId` exists
-const getById = (id: number) =>
-  db.selectFrom("Resource").where("Resource.id", "=", id)
+const getById = ({
+  resourceId,
+  siteId,
+}: {
+  resourceId: number
+  siteId: number
+}) =>
+  db
+    .selectFrom("Resource")
+    .where("Resource.id", "=", resourceId)
+    .where("siteId", "=", siteId)
```

Comment:
actually i think we don't need the site id because a resource is tied to a site by default. will we have a case where site id adds value that resource id doesn't?

Reply:
Eventually, we still require `siteId` to validate that a user has permissions to the resource imo. Just put in now also nbd